ui/desktop/src/main.ts

```diff
@@ -452,7 +452,7 @@ const getGooseProvider = () => {
 };
 
 const generateSecretKey = () => {
-  const key = crypto.randomBytes(32).toString('hex');
+  const key = process.env.GOOSE_EXTERNAL_BACKEND ? 'test' : crypto.randomBytes(32).toString('hex');
```

Comment:
maybe here you could just check if GOOSE_SERVER__SECRET_KEY is already set, and if so, use that instead of generating one. Then in the Justfile set it